src/coreclr/nativeaot/Runtime/CMakeLists.txt

```
@@ -271,7 +271,7 @@ if(CLR_CMAKE_TARGET_WIN32)
     add_definitions(-DFEATURE_SPECIAL_USER_MODE_APC)
   endif()
 else()
-  if(NOT CLR_CMAKE_TARGET_MACCATALYST AND NOT CLR_CMAKE_TARGET_IOS AND NOT CLR_CMAKE_TARGET_TVOS)
+  if(NOT CLR_CMAKE_TARGET_MACCATALYST AND NOT CLR_CMAKE_TARGET_IOS AND NOT CLR_CMAKE_TARGET_TVOS AND NOT CLR_CMAKE_TARGET_OSX)
```

Just block on all apple platforms instead of individually listing them?

```suggestion
  if(NOT CLR_CMAKE_TARGET_APPLE)
```